docs/_includes/DeveloperGuide/Appendices/planned_enhancements.md

```diff
@@ -0,0 +1,17 @@
+To address known limitations and improve user experience in SpleetWaise, the following planned enhancements have been outlined. These enhancements focus on refining existing features, improving error handling, and expanding functionality. Each item targets specific user feedback and usage scenarios to make the app more efficient and user-friendly.
+
+### Increase usability of search-related features
+    
+Search related commands such as `find` and `filterTxn` could be more intuitive by allowing partial matches and relaxing constraints on keywords. For instance, we aim to improve `find` so users can find contacts based on partial matches not just `names`, but also `phone numbers`, and `email` `addresses`, etc. Similarly, the `filterTxn` command currently supports exacts matches for filter by `amount`, `person`, `date`, `status`, and `category`; only `description` support partial matching. We plan to explore partial matching for other parameters such as the `day`, `month`, `year` in `date`, making it easier for users to find transactions with similar or related keywords.
+
+### Enhance specificity in error messages
+
+Currently, error messages may lack specificity, such as in the case of duplicate transactions - stating "Transaction already exists in the transaction book" without further details. We plan to enhance error messages to provide clear reasons for the error, helping users correct errors faster. For example, specify "Transaction already exists in the transaction book: duplicate fields with description, date and amount."
+
+### Enhance input validation
+Some inputs, like phone numbers are currently restricted to the format of Singapore's phone number, which can frustrate users if they have internation friends and want to save phone numbers of different format, such as having country code. We plan to adjust validation to allow broader input ranges where possible and potentially utilise warnings instead of outright rejection to improve user flexibility.
+
+### Implement duplicate detection improvements
+Duplicate detection in address book and transaction book can be ambiguous, as near match cases like "John Doe" and "john doe" might not be flagged correctly. We aim to refine duplicate detection to include near-match cases and provide warnings to avoid potential duplicate entries.
+
+These enhancements aim to improve usability, error handling, and functionality, enhancing the overall SpleetWaise experience for users.
```

Comment:
For me, is it possible to add these two into the planned enhancement
# Allow the Ability to Filter by Category Using the UI

Currently, users are able to use the UI to navigate filtering of the transaction list. However due to time contraints we are unable to include the filtering of category before the feature freeze of 1.6.

This improvement could be attached at the bottom of our dropdown menu of possible fields that can be filtered by clicking the currently existing filter icon on the transaction panel.

This enhancement would further improve the user-friendliness of the app where users could more easily find specific transaction relevant to the categoires they added.

# Ability to Filter Multiple Categories Simultaneously

For users with interests that span multiple categories, a single-category filter may be limiting. Enabling the ability to filter multiple categories simultaneously would allow users to customize their experience further, giving them a more powerful and flexible search tool.

This improvement could include allowing the predicate that handles the checking of categories to be able to take in varargs of category instead of just one single category

This feature would enhance the app’s flexibility and cater to a wider range of user preferences. It’s especially useful in cases where content or items may overlap categories, or where users have interests that don’t neatly fit into a single category.